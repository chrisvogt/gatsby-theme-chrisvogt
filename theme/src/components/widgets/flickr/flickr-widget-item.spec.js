import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import FlickrWidgetItem from './flickr-widget-item'

describe('FlickrWidgetItem', () => {
  const mockHandleClick = jest.fn()

  const defaultProps = {
    handleClick: mockHandleClick,
    index: 0,
    photo: {
      id: '0123456789',
      title: 'Test Photo Title',
      thumbnailUrl: 'https://cdn.chrisvogt.me/images/fake-flickr-image.jpg'
    }
  }

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('matches the snapshot', () => {
    const { asFragment } = render(<FlickrWidgetItem {...defaultProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the image with correct alt text and source', () => {
    render(<FlickrWidgetItem {...defaultProps} />)

    const img = screen.getByAltText('Flickr photo: Test Photo Title')
    expect(img).toBeInTheDocument()
    expect(img).toHaveAttribute('src', defaultProps.photo.thumbnailUrl)
  })

  it('calls handleClick with the correct index when the button is clicked', () => {
    render(<FlickrWidgetItem {...defaultProps} />)

    const button = screen.getByRole('button')
    fireEvent.click(button)

    expect(mockHandleClick).toHaveBeenCalledTimes(1)
    expect(mockHandleClick).toHaveBeenCalledWith(defaultProps.index)
  })

  it('handles missing photo data gracefully', () => {
    const propsWithMissingData = {
      handleClick: mockHandleClick,
      index: 0,
      photo: {}
    }

    render(<FlickrWidgetItem {...propsWithMissingData} />)

    const img = screen.getByRole('img')
    expect(img).toBeInTheDocument()
    expect(img).toHaveAttribute('alt', 'Flickr photo: ')
    expect(img).toHaveAttribute('src', '')
  })
})
