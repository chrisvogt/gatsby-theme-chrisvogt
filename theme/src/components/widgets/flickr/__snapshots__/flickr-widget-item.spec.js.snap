// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlickrWidgetItem matches the snapshot 1`] = `
<DocumentFragment>
  <button
    class="flickr-item-button css-1u8qly9"
  >
    <img
      alt="Flickr photo: Test Photo Title"
      class="flickr-item-image css-16gz8ax"
      crossorigin="anonymous"
      height="280"
      loading="lazy"
      src="https://cdn.chrisvogt.me/images/fake-flickr-image.jpg"
      width="280"
    />
  </button>
</DocumentFragment>
`;
